crates/bevy_ecs/src/world/reflect.rs

```diff
@@ -0,0 +1,344 @@
+//! Provides additional functionality for [`World`] when the `bevy_reflect` feature is enabled.
+
+use crate::prelude::*;
+use crate::world::ComponentId;
+use core::any::TypeId;
+
+use thiserror::Error;
+
+impl World {
+    /// Retrieves a reference to the given `entity`'s [`Component`] of the given `type_id` using
+    /// reflection.
+    ///
+    /// Requires implementing [`Reflect`] for the [`Component`] (e.g., using [`#[derive(Reflect)`](derive@bevy_reflect::Reflect))
+    /// and `app.register_type::<TheComponent>()` to have been called[^note-reflect-impl].
+    ///
+    /// If you want to call this with a [`ComponentId`], see [`World::components`] and [`Components::get_id`] to get
+    /// the corresponding [`TypeId`].
+    ///
+    /// Also see the crate documentation for [`bevy_reflect`] for more information on
+    /// [`Reflect`] and bevy's reflection capabilities.
+    ///
+    /// # Errors
+    ///
+    /// See [`GetComponentReflectError`] for the possible errors and their descriptions.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// use bevy_reflect::Reflect;
+    /// use std::any::TypeId;
+    ///
+    /// // define a `Component` and derive `Reflect` for it
+    /// #[derive(Component, Reflect)]
+    /// struct MyComponent;
+    ///
+    /// // create a `World` for this example
+    /// let mut world = World::new();
+    ///
+    /// // Note: This is usually handled by `App::register_type()`, but this example cannot use `App`.
+    /// world.init_resource::<AppTypeRegistry>();
+    /// world.get_resource_mut::<AppTypeRegistry>().unwrap().write().register::<MyComponent>();
+    ///
+    /// // spawn an entity with a `MyComponent`
+    /// let entity = world.spawn(MyComponent).id();
+    ///
+    /// // retrieve a reflected reference to the entity's `MyComponent`
+    /// let comp_reflected: &dyn Reflect = world.get_reflect(entity, TypeId::of::<MyComponent>()).unwrap();
+    ///
+    /// // make sure we got the expected type
+    /// assert!(comp_reflected.is::<MyComponent>());
+    /// ```
+    ///
+    /// # Note
+    /// Requires the `bevy_reflect` feature (included in the default features).
+    ///
+    /// [`Components::get_id`]: crate::component::Components::get_id
+    /// [`ReflectFromPtr`]: bevy_reflect::ReflectFromPtr
+    /// [`TypeData`]: bevy_reflect::TypeData
+    /// [`Reflect`]: bevy_reflect::Reflect
+    /// [`App::register_type`]: ../../bevy_app/struct.App.html#method.register_type
+    /// [^note-reflect-impl]: More specifically: Requires [`TypeData`] for [`ReflectFromPtr`] to be registered for the given `type_id`,
+    ///     which is automatically handled when deriving [`Reflect`] and calling [`App::register_type`].
+    #[inline]
+    #[cfg(feature = "bevy_reflect")]
```

Comment:
Since this entire module is gated behind the `bevy_reflect` feature, you can actually remove all these other feature checks.

Reply:
Ah yeah, I thought I caught them, but I didn't actually search for them :see_no_evil: 

Done now, removed them, thanks :+1: